internal/services/dataprotection/data_protection_backup_policy_postgresql_flexible_server_resource_test.go

```diff
@@ -0,0 +1,205 @@
+	return fmt.Sprintf(`
+provider "azurerm" {
+  features {}
+}
+
+resource "azurerm_resource_group" "test" {
+  name     = "acctest-dataprotection-%d"
+  location = "%s"
+}
+
+resource "azurerm_data_protection_backup_vault" "test" {
+  name                = "acctest-dbv-%d"
+  resource_group_name = azurerm_resource_group.test.name
+  location            = azurerm_resource_group.test.location
+  datastore_type      = "VaultStore"
+  redundancy          = "LocallyRedundant"
+}
+`, data.RandomInteger, data.Locations.Primary, data.RandomInteger)
+}
+
+func (r DataProtectionBackupPolicyPostgreSQLFlexibleServerResource) basic(data acceptance.TestData) string {
+	return fmt.Sprintf(`
+%s
+
+resource "azurerm_data_protection_backup_policy_postgresql_flexible_server" "test" {
+  name                            = "acctest-dbp-%d"
+  vault_id                        = azurerm_data_protection_backup_vault.test.id
+  backup_repeating_time_intervals = ["R/2021-05-23T02:30:00+00:00/P1W"]
+
+  default_retention_rule {
+    life_cycle {
+      duration        = "P4M"
+      data_store_type = "VaultStore"
+    }
+  }
+}
+`, r.template(data), data.RandomInteger)
+}
+
+func (r DataProtectionBackupPolicyPostgreSQLFlexibleServerResource) requiresImport(data acceptance.TestData) string {
+	return fmt.Sprintf(`
+%s
+
+resource "azurerm_data_protection_backup_policy_postgresql_flexible_server" "import" {
+  name                            = azurerm_data_protection_backup_policy_postgresql_flexible_server.test.name
+  vault_id                        = azurerm_data_protection_backup_policy_postgresql_flexible_server.test.vault_id
+  backup_repeating_time_intervals = ["R/2021-05-23T02:30:00+00:00/P1W"]
+
+  default_retention_rule {
+    life_cycle {
+      duration        = "P4M"
+      data_store_type = "VaultStore"
+    }
+  }
+}
+`, r.basic(data))
+}
+
+func (r DataProtectionBackupPolicyPostgreSQLFlexibleServerResource) complete(data acceptance.TestData) string {
+	return fmt.Sprintf(`
+%s
+
+resource "azurerm_data_protection_backup_policy_postgresql_flexible_server" "test" {
+  name                            = "acctest-dbp-%d"
+  vault_id                        = azurerm_data_protection_backup_vault.test.id
+  backup_repeating_time_intervals = ["R/2021-05-23T02:30:00+00:00/P1W"]
+  time_zone                       = "India Standard Time"
+
+  default_retention_rule {
+    life_cycle {
+      duration        = "P4M"
+      data_store_type = "VaultStore"
+    }
+  }
+
+  retention_rule {
+    name     = "weekly"
+    priority = 20
+
+    life_cycle {
+      duration        = "P6M"
+      data_store_type = "VaultStore"
+    }
+
+    criteria {
+      absolute_criteria = "FirstOfWeek"
+    }
+  }
+
+  retention_rule {
+    name     = "thursday"
+    priority = 25
+
+    life_cycle {
+      duration        = "P1W"
+      data_store_type = "VaultStore"
+    }
+
+    criteria {
+      days_of_week           = ["Thursday"]
+      months_of_year         = ["November"]
```

Comment:
shall we add a testcase for `months_of_year` with more than one element?

Reply:
Updated